pytest/executepytest.py

```diff
@@ -0,0 +1,149 @@
+
+from PythonExtensionsCollection.String.CString import CString
+from PythonExtensionsCollection.Folder.CFolder import CFolder
+
+col.init(autoreset=True)
+
+COLBR = col.Style.BRIGHT + col.Fore.RED
+COLBY = col.Style.BRIGHT + col.Fore.YELLOW
+COLBG = col.Style.BRIGHT + col.Fore.GREEN
+
+SUCCESS = 0
+ERROR   = 1
+
+# --------------------------------------------------------------------------------------------------------------
+
+def printerror(sMsg):
+    sys.stderr.write(COLBR + f"Error: {sMsg}!\n")
+
+def printexception(sMsg):
+    sys.stderr.write(COLBR + f"Exception: {sMsg}!\n")
+
+# --------------------------------------------------------------------------------------------------------------
+
+# -- some informations about the environment of this script
+
+sThisScript     = sys.argv[0]
+sThisScript     = CString.NormalizePath(sThisScript)
+sThisScriptPath = os.path.dirname(sThisScript)
+sThisScriptName = os.path.basename(sThisScript)
+
+sOSName         = os.name
+sPlatformSystem = platform.system()
+sPythonPath     = CString.NormalizePath(os.path.dirname(sys.executable))
+sPython         = CString.NormalizePath(sys.executable)
+sPythonVersion  = sys.version
+
+sFilter = None
+if sPlatformSystem == "Windows":
+    sFilter = "not _Linux_"
+elif sPlatformSystem == "Linux":
+    sFilter = "not _Windows_"
+else:
+   bSuccess = False
+   sResult  = f"Operating system {sPlatformSystem} ({sOSName}) not supported"
+   printerror(CString.FormatResult(sThisScriptName, bSuccess, sResult))
+   sys.exit(ERROR)
+
+print()
+print(f"{sThisScriptName} is running under {sPlatformSystem} ({sOSName})")
+print()
+
+# -- parse the command line of this script (optional path and name of pytest xml log file)
+
+oCmdLineParser = argparse.ArgumentParser()
+oCmdLineParser.add_argument('--logfile', type=str, help='Path and name of XML log file.')
+oCmdLineArgs = oCmdLineParser.parse_args()
+sLogFile = None
+if oCmdLineArgs.logfile is not None:
+   sLogFile = CString.NormalizePath(oCmdLineArgs.logfile, sReferencePathAbs=sThisScriptPath)
+else:
+   sLogFile = f"{sThisScriptPath}/logfiles/PyTestLog.xml"
+
+# -- create the log file folder
+
+sLogFilesPath = os.path.dirname(sLogFile)
+oLogFilesPath = CFolder(sLogFilesPath)
+bSuccess, sResult = oLogFilesPath.Create(bOverwrite=False, bRecursive=True)
+del oLogFilesPath
+if bSuccess is not True:
+   printerror(CString.FormatResult(sThisScriptName, bSuccess, sResult))
+   sys.exit(ERROR)
+print(sResult)
+print()
+
+# -- prepare the command line for the test execution
+
+listCmdLineParts = []
+listCmdLineParts.append(f"\"{sPython}\"")
+listCmdLineParts.append("-m pytest")
+if sFilter is not None:
+   listCmdLineParts.append(f"-k \"{sFilter}\"")
+listCmdLineParts.append("--show-capture=all")
+listCmdLineParts.append(f"--junitxml=\"{sLogFile}\"")
+listCmdLineParts.append(f"\"{sThisScriptPath}\"")
+sCmdLine = " ".join(listCmdLineParts)
+del listCmdLineParts
+
+# -- execute the tests
+
+listCmdLineParts = shlex.split(sCmdLine)
```

Comment:
Hi Holger,
hm... I'm not sure if this is elegant to fill first an array, then join it and then split it again?!?
But I have no nicer proposal. The optional "sFilter" makes it difficult to create an anonymous array or directly a string...
Thank you,
 Thomas

Reply:
Hi Thomas,

it is like it is. I start with a list because in case of optional parts this is easier to handle than concatenating strings.
I need to join because of shlex.split requires a string - and returns a list. The second join (of this returned list)
in line 125 I only do to give the user a feedback about what will be executed. But I want to see what shlex has done
with my input because this is what will be executed and therefore I do not use the already merged original list for feedback.